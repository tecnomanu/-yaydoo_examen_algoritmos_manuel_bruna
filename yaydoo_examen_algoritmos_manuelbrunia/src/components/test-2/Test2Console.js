import React from 'react'

export const Test2Console = ( {words, str} ) => {
    const keys = ["{", "}"]
    return (
        <>
            <strong>Console:</strong><br />
            <code>Input string: "{str}"</code><br /><br />
            <code>Output:  {keys[0]}
                {
                    Object.keys(words).map((key, index) => {
                        return (`"${key}" : ${words[key]}${(index < Object.keys(words).length - 1 ? ', ' : '')}`)
                    })
                }
                {keys[1]}
            </code>
        </>
    )
}
