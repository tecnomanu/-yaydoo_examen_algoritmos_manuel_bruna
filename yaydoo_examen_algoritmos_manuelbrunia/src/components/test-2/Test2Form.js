import React from 'react'
import { useForm } from '../../hooks/useForm/useForm';

export const Test2Form = ({defaultInput, setStringState}) => {
        
    const [values, handleInputChange] = useForm({
        stringInput: defaultInput
    });

    const {stringInput} = values;

    const handleSearch = (e)=>{
        e.preventDefault();
        setStringState(stringInput);
    };

    return (
        <form onSubmit={handleSearch}>
            <textarea className="form-control"
                name="stringInput"
                value={stringInput}
                onChange={handleInputChange}
                rows="10"
                placeholder="Pegue el texto aquÃ­ para contar sus palabras...">
            </textarea>
            <button type="submit"
                className="btn btn-primary mt-2">
                Contar Palabras
            </button>
        </form>
    )
}
