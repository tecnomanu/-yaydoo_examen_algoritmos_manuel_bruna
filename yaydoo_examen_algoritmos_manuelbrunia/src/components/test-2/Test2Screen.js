import React, { useMemo, useState, useEffect } from 'react'
import { Test2Console } from './Test2Console';
import { Test2Form } from './Test2Form';
import { getCountWords } from '../../helpers/getCountWord';

export const Test2Screen = React.memo(() => {
    
    const string_default = localStorage.getItem('string_count') || `Érase una vez una niñita que lucía una hermosa capa
    de color rojo. Como la niña la usaba muy a menudo, todos la
    llamaban Caperucita Roja. Un día, la mamá de Caperucita Roja la
    llamó y le dijo: —Abuelita no se siente muy bien, he horneado
    unas galletitas y quiero que tú se las lleves. —Claro que sí
    —respondió Caperucita Roja, poniéndose su capa y llenando su
    canasta de galletas recién horneadas. Antes de salir, su mamá
    le dijo: — Escúchame muy bien, quédate en el camino y nunca
    hables con extraños. —Yo sé mamá —respondió Caperucita Roja y
    salió inmediatamente hacia la casa de la abuelita.`;
    
    const [string, setStringState] = useState(string_default);
    const [words, setWordsState] = useState(string);

    useEffect(() => {
        localStorage.setItem('string_count', string);
        setWordsState(getCountWords(string));
    }, [string])

    return (
        <div className="container">
            <div className="row">
                <div className="col-md-12  pt-2">
                    <h1 className="text-center">Prueba 2 - Cuentapalabras</h1>
                    <p>Este cuenta palabras tiene la posibilidad de cambiar el texto actual por el que nosotros querramos, para poder contar las palabras de forma dinamica.</p>
                    <Test2Form 
                        defaultInput={string_default}
                        setStringState={setStringState} />
                    <strong>Si usamos el inspector, en la consola se imprime el objecto con el contador.</strong>
                </div>
                <div className="col-md-12 background-code mt-2 pt-4 pb-4">
                    <Test2Console words={words} str={string} />
                </div>
            </div>
        </div>
    )
})