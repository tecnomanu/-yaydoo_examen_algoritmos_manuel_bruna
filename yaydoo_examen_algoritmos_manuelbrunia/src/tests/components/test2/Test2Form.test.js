import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { Test2Form } from '../../../components/test-2/Test2Form';

describe("Test on <Test2Form/>", ()=>{

    const setStringState = jest.fn();
    const default_string = "hola Hola Mundo";
    const wrapper = shallow(<Test2Form defaultInput={default_string} setStringState={setStringState}/>);

    test('SnapShot of <Test2Form/>', ()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test('The button must call setStringState with string when has been clicked', ()=>{
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setStringState ).toHaveBeenCalledTimes(1)
        expect( setStringState ).toHaveBeenCalledWith(default_string);
    })

    test('Change string on textarea', ()=>{
        const wrapper_to_change = shallow(<Test2Form defaultInput={default_string} setStringState={setStringState}/>);
        const string_to_change = "Lorem lorem ipsum";
        wrapper_to_change.find('textarea').simulate('change', {
            target: {
                value: string_to_change,
                name: "stringInput"
            }
        });
        expect( wrapper_to_change.find('textarea').prop('value') ).toBe(string_to_change);
    })

})