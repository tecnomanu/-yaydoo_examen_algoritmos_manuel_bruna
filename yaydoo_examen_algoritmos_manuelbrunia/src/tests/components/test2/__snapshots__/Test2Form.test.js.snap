// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Test2Form/> SnapShot of <Test2Form/> 1`] = `
<form
  onSubmit={[Function]}
>
  <textarea
    className="form-control"
    name="stringInput"
    onChange={[Function]}
    placeholder="Pegue el texto aquÃ­ para contar sus palabras..."
    rows="10"
    value="hola Hola Mundo"
  />
  <button
    className="btn btn-primary mt-2"
    type="submit"
  >
    Contar Palabras
  </button>
</form>
`;
